crates/compiler/builtins/roc/Bool.roc

```diff
@@ -38,67 +44,67 @@ true = @Bool True
+## the logic [AND](https://en.wikipedia.org/wiki/Logical_conjunction) 
+## gate. The infix operator `&&` can also be used as shorthand for 
+## `Bool.and`.
+##
+##     expect (Bool.and Bool.true Bool.true) == Bool.true
+##     expect (Bool.true && Bool.true) == Bool.true
+##     expect (Bool.false && Bool.true) == Bool.false
+##     expect (Bool.true && Bool.false) == Bool.false
+##     expect (Bool.false && Bool.false) == Bool.false
+##
+## **Performance Note** that in Roc the `&&` and `||` work the same way as any 
+## other function. However, in some languages `&&` and `||` are special-cased.
+## In these languages the compiler will skip evaluating the expression after the
+## first operator under certain circumstances. For example an expression like
+## `enablePets && likesDogs user` would compile to.
 ##
 ##     if enablePets then
 ##         likesDogs user
 ##     else
 ##         Bool.false
 ##
-## In Roc, however, `&&` and `||` are not special. They work the same way as
-## other functions. Conditionals like `if` and `when` have a performance cost,
-## and sometimes calling a function like `likesDogs user` can be faster across
-## the board than doing an `if` to decide whether to skip calling it.
-##
-## (Naturally, if you expect the `if` to improve performance, you can always add
-## one explicitly!)
+## Roc does not do this because conditionals like `if` and `when` have a 
+## performance cost. Calling a function can sometimes be faster across the board
+## than doing an `if` to decide whether to skip calling it.
 and : Bool, Bool -> Bool
 
-## Returns `Bool.true` when given `Bool.true` for either argument, and `Bool.false` only when given `Bool.false` and `Bool.false`.
-##
-## `a || b` is shorthand for `Bool.or a b`.
-##
-## >>> Bool.true || Bool.true
-##
-## >>> Bool.true || Bool.false
-##
-## >>> Bool.false || Bool.true
-##
-## >>> Bool.false || Bool.false
-##
-## ## Performance Notes
-##
-## In some languages, `&&` and `||` are special-cased in the compiler to skip
-## evaluating the expression after the operator under certain circumstances.
-## In Roc, this is not the case. See the performance notes for [Bool.and] for details.
+## Returns `Bool.true` when either input is a `Bool.true`. This is equivalent to
+## the logic [OR](https://en.wikipedia.org/wiki/Logical_disjunction) gate.
+## The infix operator `||` can also be used as shorthand for `Bool.or`.
+##
+##     expect (Bool.or Bool.false Bool.true) == Bool.true
+##     expect (Bool.true || Bool.true) == Bool.true
+##     expect (Bool.false || Bool.true) == Bool.true
+##     expect (Bool.true || Bool.false) == Bool.true
+##     expect (Bool.false || Bool.false) == Bool.false
+##
+## **Performance Note** that in Roc the `&&` and `||` work the same way as any 
+## other functions. However, in some languages `&&` and `||` are special-cased. 
+## Refer to the note in `Bool.and` for more detail.
 or : Bool, Bool -> Bool
-# xor : Bool, Bool -> Bool # currently unimplemented
```

Comment:
Note the `Bool.isNotEq` is equal to a XOR gate, so I removed this as it is not needed.